sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -396,6 +400,102 @@ def test_davies_bouldin_score():
     pytest.approx(davies_bouldin_score(X, labels), (5.0 / 4) / 3)
 
 
+@pytest.fixture
+def density_sample():
+    # start off with two non-spherical clusters
+    points, labels = datasets.make_moons()
+    bounds = np.array([np.max(points, axis=0), np.min(points, axis=0)]).transpose()
+    # add one point labelled as noise in each "corner"
+    points = np.append(points, list(itertools.product(*bounds)), axis=0)
+    labels = np.append(labels, [-1 for _ in range(4)])
```

Comment:
I decided to place the noise points in the corners since that seemed to align more with the way an actual clustering algorithm would assign noise. (Which should represent the overwhelming majority of scenarios the scoring function will be used in)

But sure, coming from a ground-truth perspective it would make sense for the noise to be uniformely distributed. I'll switch to generation via `np.random.uniform`.

Reply:
18dc1769318905aea107b8e5984628151d5fc47f